tests/test_fcm.py

```
@@ -0,0 +1,359 @@
+    async def test_listener_does_not_record_message_when_ack_fails(self) -> None:
+        module = self.module
+
+        class FailingAckWriter(FakeWriter):
+            def __init__(self) -> None:
+                super().__init__()
+                self.drain_count = 0
+
+            async def drain(self) -> None:
+                self.drain_count += 1
+                if self.drain_count == 2:
+                    raise ConnectionResetError("ack failed")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)